vast/vastutil/validate.go

```diff
@@ -0,0 +1,39 @@
+package vastutil
+
+import (
+	"encoding/xml"
+	"fmt"
+	"strings"
+
+	"github.com/Vungle/vungo/vast/vastelement"
+)
+
+// Validate method validate VAST data as an external api
+func Validate(vasts []*vastelement.Vast, version vastelement.Version) error {
+	var impressionsErrCount int
+	var hasLinear bool
+	for _, v := range vasts {
+		if err := v.Validate(version); err != nil {
+			invalidVast, marshalErr := xml.Marshal(v)
+			if marshalErr != nil {
+				return fmt.Errorf("failed to marshal VAST for logging: %v", marshalErr)
+			}
+			if strings.Contains(err.Error(), vastelement.ErrWrapperMissImpressions.Error()) &&
+				impressionsErrCount < len(vasts)-1 {
+				impressionsErrCount++
+			} else {
+				return fmt.Errorf("failed to validate VAST: %v, VAST: %s", err.Error(), invalidVast)
```

Comment:
If there is any error except ErrWrapperMissImpressions, the validation is finished immediately. Otherwise, validator should count ErrWrapperMissImpressions and judge whether impressionsErrCount equals len(vasts)-1, if the two are equal, that means  impressionsErrCount will equal  len(vasts) after the final loop. The code is not very graceful, but it is correct.